packages/language-server/src/plugins/typescript/service.ts

```diff
@@ -318,6 +325,27 @@ async function createLanguageService(
         reduceLanguageServiceCapabilityIfFileSizeTooBig();
     }
 
+    function getScriptFileNames() {
+        const projectFiles = languageServiceReducedMode
+            ? []
+            : snapshotManager.getProjectFileNames();
+        const canonicalProjectFileNames = projectFiles.map(getCanonicalFileName);
+
+        return Array.from(
+            new Set([
+                ...projectFiles,
+                // project file is read from the file system so it's more likely to have
+                // the correct casing
+                ...snapshotManager
+                    .getFileNames()
+                    .filter(
+                        (file) => !canonicalProjectFileNames.includes(getCanonicalFileName(file))
```

Comment:
I believe this method is called rather often, and this would make it O(n^2) - can we use a `Set` instead?